### DISCUSSION TITLE
Planned support for word-level n-gram LM with streaming decoder?

### DISCUSSION THREAD

**Discussion by agne-ker** [espnet/espnet#4538] [Q&A] [answered]
I want to be able to decode with a subword-level ESPnet model plus a word-level n-gram LM. Is that supported in ESPnet? I am specifically interested in the use with the streaming decoder, but I cannot find any mention of n-gram LMs in the code...Is there any plans to implement word-level n-grams for such a use case?

**Comment by sw005320** [comment, marked as answer]
There are some research projects but we are not planning to support them in the official repo since we found that BPE-token-based LMs are good enough for many benchmarks, if we have a sufficient amount of data.

**Reply by agne-ker** [reply]
Thanks for your answer.
